web/satellite/src/store/modules/usage.ts

```diff
@@ -1,78 +1,92 @@
-    },
-    mutations: {
-        [PROJECT_USAGE_MUTATIONS.FETCH](state: any, projectUsage: ProjectUsage) {
-           state.projectUsage = projectUsage;
-        },
-        // TODO: create type here instead of {startDate, endDate}
-        [PROJECT_USAGE_MUTATIONS.SET_DATE](state: any, {startDate, endDate}: any) {
-            state.startDate = startDate as Date;
-            state.endDate = endDate as Date;
-        },
-        [PROJECT_USAGE_MUTATIONS.CLEAR](state:any) {
-            state.projectUsage = {storage: 0, egress: 0, objectCount: 0} as ProjectUsage;
-            state.startDate = new Date();
-            state.endData = new Date();
-        }
-    },
-    actions: {
-        [PROJECT_USAGE_ACTIONS.FETCH]: async function({commit, rootGetters}: any, {startDate, endDate}: any): Promise<RequestResponse<ProjectUsage>> {
-            const projectID = rootGetters.selectedProject.id;
-
-            let result: RequestResponse<ProjectUsage> = await fetchProjectUsage(projectID, startDate, endDate);
-
-            if (result.isSuccess) {
-                commit(PROJECT_USAGE_MUTATIONS.SET_DATE, {startDate, endDate});
-                commit(PROJECT_USAGE_MUTATIONS.FETCH, result.data);
-            }
+import { ProjectUsageApiGql } from '@/api/usage';
+import { StoreModule } from '@/store';
+import { ProjectUsage, DateRange } from '@/types/usage';
 
-            return result;
-        },
-        [PROJECT_USAGE_ACTIONS.FETCH_CURRENT_ROLLUP]: async function({commit, rootGetters}: any): Promise<RequestResponse<ProjectUsage>> {
-            const projectID: string = rootGetters.selectedProject.id;
+export const PROJECT_USAGE_ACTIONS = {
+    FETCH: 'fetchProjectUsage',
+    FETCH_CURRENT_ROLLUP: 'fetchCurrentProjectUsage',
+    FETCH_PREVIOUS_ROLLUP: 'fetchPreviousProjectUsage',
+    CLEAR: 'clearProjectUsage',
+};
 
-            const endDate = new Date();
-            const startDate = new Date(Date.UTC(endDate.getUTCFullYear(), endDate.getUTCMonth(), 1));
+export const PROJECT_USAGE_MUTATIONS = {
+    FETCH: 'FETCH_PROJECT_USAGE',
+    SET_DATE: 'SET_DATE_PROJECT_USAGE',
+    CLEAR: 'CLEAR_PROJECT_USAGE'
+};
 
-            let result: RequestResponse<ProjectUsage> = await fetchProjectUsage(projectID, startDate, endDate);
+const defaultState = new ProjectUsage(0, 0, 0, new Date(), new Date());
 
-            if (result.isSuccess) {
-                commit(PROJECT_USAGE_MUTATIONS.SET_DATE, {startDate, endDate});
-                commit(PROJECT_USAGE_MUTATIONS.FETCH, result.data);
-            }
+class UsageState {
+    public projectUsage: ProjectUsage = defaultState;
+    public startDate: Date = new Date();
+    public endDate: Date = new Date();
+}
 
-            return result;
+export function makeUsageModule(api: ProjectUsageApiGql): StoreModule<UsageState> {
+    return {
+        state: new UsageState(),
+        mutations: {
+            [PROJECT_USAGE_MUTATIONS.FETCH](state: UsageState, projectUsage: ProjectUsage) {
```

Comment:
Should be named like ``SET_PROJECT_USAGE``. Mutations doesn't perform any ``FETCH``, It can only set, delete, or modify